packages/client/src/debounce.test.tsx

```diff
@@ -0,0 +1,56 @@
+import debounce from "./debounce"
+
+jest.useFakeTimers()
+
+describe("debounce", () => {
+  let func: jest.Mock
+  let debouncedFunc: Function
+
+  beforeEach(() => {
+    func = jest.fn()
+    debouncedFunc = debounce(func, 3000, true)
+  })
+
+  it("executes once over a period shorter than 3 seconds", () => {
+    debouncedFunc()
+    expect(func).toBeCalledTimes(1)
+
+    for (let i = 0; i < 300; i++) {
+      debouncedFunc()
+    }
+
+    jest.runAllTimers()
+
+    expect(func).toBeCalledTimes(1)
+  })
+
+  it("calls func every 3 seconds over a period longer than 3 seconds", () => {
+    debouncedFunc()
+    expect(func).toBeCalledTimes(1)
+
+    for (let i = 0; i < 10; i++) {
+      debouncedFunc()
+      jest.advanceTimersByTime(3000)
+    }
+
+    expect(func).toBeCalledTimes(10)
+  })
+
+  it("eventually calls func with the same argument as the last call to debouncedFunc", () => {
+    let i
+    let args: number[] = []
```

Comment:
You only really care about the last argument, so you don't need an array here. If you just track the latest argument then `i` can be scoped to only the `for` block.